### PR TITLE
feat: Initial django setup

### LINKED ISSUE #2164: V2-Setup Django in the project
**Is your feature request related to a problem? Please describe.**
Setup django in the backend folder of the project on v2 Branch

**Describe the solution you'd like**
With the following discussion, it has been decided to reboot the badgeyay with ReactJS+Django

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #2164 

#### Short description of what this resolves:
Setup django initially for the project.

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal please enable codacy, it's not running currently. Thanks

**Review by iamareebjamal — COMMENTED** [review]
App name should not be backend. It should be badgeyay

**Comment by codedsun** [comment]
> App name should not be backend. It should be badgeyay

The root folder name created by django should be backend or same as app name?

File: backend/backend/settings.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+WSGI_APPLICATION = 'backend.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/2.2/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
```
Comment (iamareebjamal) [inline backend/backend/settings.py]: Change to postgres

Wait till the requirements of the project are finalized

**Comment by iamareebjamal** [comment]
backend/badgeyay

File: backend/backend/settings.py (hunk 1 above)
Comment (codedsun) [reply backend/backend/settings.py]: Done! Ok

**Comment by mariobehling** [comment]
The requirements of the project are quite basic. For now:
* implement a way to create and store badges similar to current system
* also user accounts with standard feature should be possible
* in further future a way to create paid features

Frontend should be in ReactJS.

**Comment by iamareebjamal** [comment]
@mariobehling If the goals are that simple, why create a separate frontend (react). We don't have any dynamic component, so a server side rendered site will be better than a client-side rendered site. 

Difference can be seen here. 

Client Side Rendered - https://open-event-fe.netlify.com/

1. Loads in 15 seconds
2. Blank page. Downloads 8 MB of Javascript
3. Then calls the API and renders the HTML. User has to wait for 15 seconds
4. The standard authentication flow is broken. User is logged out of every 24 hours and is more difficult to integrate than normal Django Auth Flow

Server Side Rendered - https://www.eventbrite.com/

1. Loads in 2 seconds
2. HTML and CSS right off the bat for the user. User can see the site as soon as HTML gets parsed. Javascript is there just to enhance the site
3. Standard authentication flow in Django. So, no duplication of authentication/authorization in the backend and frontend. All auth lies in the server, so no bugs like a speaker can access admin panel because there was a bug in frontend. Or speaker can override email because there was a bug in backend. All logic remains in 1 place.

It can be said that open-event-server was a complex piece of software (frontend wise), so it made sense to separate its backend and frontend, but eventbrite is as complex if not more and they have done ~~fine~~ better with Django SSR and then JavaScript for enhancement. And badgeyay is even simpler. Django only will give a lot of benefits and an API can be made as a single source of truth for other clients.

All great websites follow the same pattern:
1. GitHub - Rails mainly and jQuery to enhance
2. GitLab - Rails + Vue to enhance
3. Eventbrite - Django + React/jQuery to enhance (Create Event + Dashboards in JS)
4. Meetup - Node.JS rendered React site

So, is there any specific reason we want to separate backend from frontend?

File: backend/badgeyay/settings.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+WSGI_APPLICATION = 'badgeyay.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/2.2/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': os.path.join(BASE_DIR, 'badgeyay'),
```
Comment (iamareebjamal) [inline backend/badgeyay/settings.py]: Just badgeyay

Comment (codedsun) [reply backend/badgeyay/settings.py]: ok!

**Comment by codedsun** [comment]
@iamareebjamal Check now

File: backend/badgeyay/settings.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+WSGI_APPLICATION = 'badgeyay.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/2.2/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': env.str('BADGEYAY_DATABASE', default='badgeyay'),
```
Comment (iamareebjamal) [inline backend/badgeyay/settings.py]: This can be managed by `settings_local.py`

Comment (iamareebjamal) [reply backend/badgeyay/settings.py]: https://djangostars.com/blog/configuring-django-settings-best-practices/

Comment (iamareebjamal) [inline backend/badgeyay/settings.py]: Where is env defined?

Comment (codedsun) [reply backend/badgeyay/settings.py]: @iamareebjamal Sorry, should I take from env or just keep it default ?

Comment (iamareebjamal) [reply backend/badgeyay/settings.py]: Just badgeyay

**Comment by codedsun** [comment]
@iamareebjamal check now

File: backend/badgeyay/settings_local.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Database
```
Comment (iamareebjamal) [inline backend/badgeyay/settings_local.py]: The point of settings_local.py itself is to not be checked into version control

File: backend/badgeyay/settings.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/2.2/howto/static-files/
+
+STATIC_URL = '/static/'
+
+try:
+    from .settings_local import *
+except ImportError:
+    pass
```
Comment (iamareebjamal) [inline backend/badgeyay/settings.py]: I meant to say leave the env setting for now, so remove these lines

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**